sql/partition_info.cc

```diff
@@ -3495,6 +3484,24 @@ bool partition_info::vers_trx_id_to_ts(THD* thd, Field* in_trx_id, Field_timesta
   return false;
 }
 
+bool partition_info::vers_interval_exceed(my_time_t max_time, partition_element *part)
+{
+  DBUG_ASSERT(vers_info);
+  if (vers_info->interval.empty())
+    return false;
+  if (!part)
+  {
+    DBUG_ASSERT(vers_info->initialized());
+    part= vers_hist_part();
+  }
+  my_time_t min_time= vers_pruning_stat(Vers_pruning_stat::ROW_END, part).min_time();
+  MYSQL_TIME max;
+  MYSQL_TIME min;
+  current_thd->variables.time_zone->gmt_sec_to_TIME(&max, max_time);
+  current_thd->variables.time_zone->gmt_sec_to_TIME(&min, min_time);
+  date_add_interval(&min, vers_info->get_interval_type(), vers_info->interval);
```

Comment:
2. Overload `date_add_interval (MYSQL_TIME *, interval_type, ulong)`. You will have to make one ugly fork for INTERVAL constructor by `interval_type` + `ulong`.

Reply:
In `sql_yacc.yy` I will translate `INTERVAL` to `interval_type + ulong`. Then here, in `partition_info::vers_interval_exceed` `interval_type + ulong` will be converted back to `INTERVAL` if I understood you correctly. What's the point in converting `INTERVAL` to `ulong`?